File: sample/sample/AdhesionViewController.m
//
//  AdhesionViewController.m
//  sample
//
//  Created by Vladas Drejeris on 04/05/15.
//  Copyright (c) 2015 Adform. All rights reserved.
//

#import "AdhesionViewController.h"

@interface AdhesionViewController () <AFAdInlineDelegate>

@property (nonatomic, strong) AFAdHesion *adHesion;

@property (nonatomic, weak) IBOutlet NSLayoutConstraint *labelTopConstraint;
@property (nonatomic, weak) IBOutlet NSLayoutConstraint *labelBottomConstraint;

@end

@implementation AdhesionViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    // Do any additional setup after loading the view.
    
    // Create and setu the ad view.
    self.adHesion = [[AFAdHesion alloc] initWithMasterTagId:[self masterTag] position:self.adPosition presentingViewController:self];
    
    if (self.adPosition == AFAdPositionTop) {
        self.adHesion.delegate = self;
    } else {
        self.adHesion.hidesOnSwipe = true;
    }

    // This line of code enables multiple ad size support.
    self.adHesion.additionalDimmensionsEnabled = true;
    
    // Add it to view hierarchy and initiate ad loading.
    [self.view addSubview:self.adHesion];
    [self.adHesion loadAd];
}

- (NSInteger )masterTag {
    
    return 142493;
}

- (void)updateViewConstraints {
    if (self.adHesion.position == AFAdPositionTop && (self.adHesion.state == AFAdStateInShowTransition || self.adHesion.state == AFAdStateVisible)) {
        self.labelTopConstraint.constant = self.adHesion.adSize.height;
    } else {
        self.labelTopConstraint.constant = 8;
    }
    
    [super updateViewConstraints];
}

#pragma mark - AFAdInlineDelegate

- (void)adInlineWillShow:(AFAdInline *)adInline {
    [self.view setNeedsUpdateConstraints];
    
    [UIView animateWithDuration:AFAdViewAnimationDuration
                     animations:^{
                         [self.view layoutIfNeeded];
                     }];
}

- (void)adInlineWillHide:(AFAdInline *)adInline {
    [self.view setNeedsUpdateConstraints];
    
    [UIView animateWithDuration:AFAdViewAnimationDuration
                     animations:^{
                         [self.view layoutIfNeeded];
                     }];
}

@end

File: sample/sample/Placements/AdhesionViewController.swift
//
//  AdhesionViewController.swift
//  sample
//
//  Created by Laurynas Letkauskas on 2021-02-26.
//

import AdformAdvertising
import UIKit

class AdhesionViewController: UIViewController {
    @IBOutlet private weak var labelTopConstraint: NSLayoutConstraint!
    @IBOutlet private weak var labelBottomConstraint: NSLayoutConstraint!
    
    var adPosition: AFAdPosition!
    private var adHesion: AFAdHesion?

    override func viewDidLoad() {
        super.viewDidLoad()
        // Do any additional setup after loading the view.

        // Create and setup the ad view.
        adHesion = AFAdHesion(masterTagId: masterTag(), position: adPosition, presenting: self)
        //adHesion.videoSettings.closeButtonBehavior = .allways
        //adHesion.videoSettings.controlsStyle = .minimal

        if adPosition == .top {
            adHesion?.delegate = self
        } else {
            adHesion?.hidesOnSwipe = true
        }

        // This line of code enables multiple ad size support.
        adHesion?.areAditionalDimmensionsEnabled = true

        // Add it to view hierarchy and initiate ad loading.
        if let adHesion = adHesion {
            view.addSubview(adHesion)
        }
        adHesion?.loadAd()
    }

    func masterTag() -> Int {

        return 987051
    }

    override func updateViewConstraints() {
        if adHesion?.position == .top && (adHesion?.state == AFAdState.inShowTransition || adHesion?.state == AFAdState.visible) {
            labelTopConstraint.constant = adHesion?.adSize.height ?? 0.0
        } else {
            labelTopConstraint.constant = 8
        }

        super.updateViewConstraints()
    }
}

extension AdhesionViewController: AFAdInlineDelegate {
    func adInlineWillShow(_ adInline: AFAdInline) {
        view.setNeedsUpdateConstraints()
        UIView.animate(withDuration: TimeInterval(AFAdViewAnimationDuration)) {
            self.view.layoutIfNeeded()
        }
    }
    
    func adInlineWillHide(_ adInline: AFAdInline) {
        view.setNeedsUpdateConstraints()
        UIView.animate(withDuration: TimeInterval(AFAdViewAnimationDuration)) {
            self.view.layoutIfNeeded()
        }
    }
}
